### ISSUE TITLE
(Bug?) Inventory Panel & inventory control 'accuracy issues'

### ISSUE THREAD

**Issue by Vaygrim** [SleepyTrousers/EnderIO#2755]
VERSIONS!
Ender IO 1.7.10 - 2.3.0.407_beta
EnderCore 1.7.10 - 0.0.1.17_alpha
MC 1.7.10 (duh)
Forge 1448

THE ISSUE!
   This problem has been happening on ForgeCraft 2, so I can't give a world download for testing. BUT there is video on my channel of how I have my setup configured.

   The issue has to do specifically with the Inventory Panel. The more inventory I have it 'control', the more I seem to lose things in the system. Currently if I let any stuff from IC2 or MFR get stored in the 'general storage' chests.. they basically get lost in the system (meaning that they will NOT show up by targeted search in the Inventory Panel and I have to go find them in their chests manually). In an attempt to 'fix' this issue I made 1 chest that was specifically filtered for IC2, MFR and Immersive Engineering parts. That corrected the issue completely.

   Until today. Today I started making more 'new' MFR parts and once I put them into the Inventory Panel.. they immediately 'vanished into the system'. I could not search for them and had to go find them 'manually'. I expected this was just an issue of the system not 'polling for an update' (or however you want to phrase it) of its' inventory and stored items. I left to do other tasks for 20 minutes and came back, but the issue persisted. I still cannot find MFR Plastic of any form in my system because of this odd problem.

  Shortly after this started happening I added some more Copper ingots to my system (I had just come back from mining for that specifically) then went to work on thaumcraft research. I wanted to search for all "Shards" from Thaumcraft and pull them out. First two types I pulled out of the inventory panel fine. The third type of shard (Earth) turned into 16 copper ingots once I got it out of the panel and onto my pointer. The fourth and fifth types of shards I then trying to remove from the system turned into Copper and Silver ingots.

I can produce example video footage of this in action if so required. Just let me know what you need.

TL;DR - I've got some items 'vanishing' into the Inventory Panel's storage system unless I go find them in the attached chests manually. Other things yet are showing up as the wrong type of item in the Inventory Panel itself. (Copper Ingots showing up as Thaumcraft shards that I actually don't HAVE anywhere in my inventory system.)

HALP! (And thanks!)

**Comment by tterrag1098** [comment]
Could you try this again with the latest beta?

**Comment by Vaygrim** [comment]
Okay so we've rolled out the updated version onto the server, and all the things I *was* losing.. are now found!

Trouble is now other things are being lost periodically.

At present I will occasionally search for "Copper" and see 3 types of Copper Ingots (as in .. from 3 different mods) in my system despite the fact that I know I only have 1 type in my system. So I pull out the 2nd and 3rd ones with smaller amounts and once on my cursor they magically transform into say.. Copper Wiring. Or Copper Plates.

If I put these items back into the system.. they transform back into copper ingots from other mods again. Really strange.

Please let me know if you need anymore testing. I would love to give you a map download so you could test it but unfortunately I don't have access to do that. I can do testing however.

**Comment by HenryLoenwind** [comment]
for reference, another report: https://www.reddit.com/r/feedthebeast/comments/3zatqt/infinity_evolved_expert_enderio_inventory_panel/

**Comment by tterrag1098** [comment]
Ref #3129

**Comment by charlesbevans** [comment]
I had the same problem in Infinity Evolved Expert. A number of things disappeared and some reappeared as I messed with the setup. In particular the IC2 plates seemed to all refuse to be recognized and resisted being fixed. Replacing the panel fixed everything.

**Comment by MatthiasMann** [comment]
We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

**Comment by charlesbevans** [comment]
It's been a little while since I fixed it. I'll grab the logs if/when it repeats.

**Comment by jmwierzbicki** [comment]
same happens to me, 20% of items inserted into system are displayed as wrong item - there is no rule, sometimes barley is displayed as tic lumberaxe, and seeds, chipsets as sticks, could not find rule.
its happening on ftb infinity 
https://i.imgur.com/gHaNjtO.png hover
https://i.imgur.com/M45Zf8D.png on click

after breaking and placing panel again, items seems to be shuffled - another items are hiding under lumber axes, and so on. My system is connected to array of storage drawers via drawer controller - but i dont think its related, because logistics pipes and AE2 has no such issues when connected to same controller at the same time.

for now, i have logistics pipes request table under panel, as backup, if something is lost in ender-io panel.

**Comment by HenryLoenwind** [comment]
We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.

**Comment by HenryLoenwind** [comment]
And in case it's not clear:

# **We need the debug output from the version linked in #3129 otherwise we can't fix this issue as we were not able to reproduce these in any of our tests.
**
